pipelines/build/common/config_regeneration.groovy

```diff
@@ -0,0 +1,433 @@
+    /*
+    * Stage: Check that the pipeline isn't in inprogress or queued up. Once clear, run the regeneration job
+    */
+    context.stage("Check $javaVersion pipeline status") {
+
+      // Get all pipelines
+      def getPipelines = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/api/json?tree=jobs[name]&pretty=true&depth1")
+
+      // Parse api response to only extract the relevant pipeline
+      getPipelines.jobs.name.each{ pipeline -> 
+        if (pipeline.contains("pipeline") && pipeline.contains(javaVersion)) {
+          Integer sleepTime = 900
+          Boolean inProgress = true
+
+          while (inProgress) {
+            // Check if pipeline is in progress using api
+            context.println "[INFO] Checking if ${pipeline} is running..." //i.e. openjdk8-pipeline
+
+            def pipelineInProgress = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/job/${pipeline}/lastBuild/api/json?pretty=true&depth1")
+            inProgress = pipelineInProgress.building as Boolean
+
+            if (inProgress) {
+              // Sleep for a bit, then check again...
+              context.println "[INFO] ${pipeline} is running. Sleeping for ${sleepTime} seconds while waiting for ${pipeline} to complete..."
+              context.sleep sleepTime
+            }
+
+          }
+          
+          context.println "[SUCCESS] ${pipeline} is idle. Running regeneration job..."
+        }
+
+      }
+
+    } // end stage
+
+    /*
+    * Stage: Regenerate all of the job configurations by job type (i.e. jdk8u-linux-x64-hotspot
+    * jdk8u-linux-x64-openj9, etc.)
+    */
+    context.stage("Regenerate $javaVersion pipeline jobs") {
+
+      // Get downstream job folder and platforms
+      Map<String,List> downstreamJobs = [:]
+
+      // i.e. jdk11u, jdk8u, etc.
+      context.println "[INFO] Pulling downstream folders and jobs from API..."
+      def folders = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/api/json?tree=jobs[name]&pretty=true&depth=1")
+
+      folders.jobs.name.each { folder -> 
+        if ((folder == "jdk" && javaVersion == "jdk15") || folder.contains(javaVersion)) {
+          def jobs = [] // clean out array each time to avoid duplication
+
+          // i.e. jdk8u-linux-x64-hotspot, jdk8u-mac-x64-openj9, etc.
+          def platforms = queryJenkinsAPI("https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/${folder}/api/json?tree=jobs[name]&pretty=true&depth=1")
+
+          platforms.jobs.name.each { job -> 
+            jobs.add(job)
+          }
+
+          downstreamJobs.put(folder, jobs)
+        }
+
+      }
+
+      // Output for user verification
+      context.println "[INFO] Jobs to be regenerated (they should only be $javaVersion jobs!):"
+      downstreamJobs.each { folder, jobs -> context.println "${folder}: ${jobs}\n" }
+
+      // Regenerate each job, running through map a job at a time
+      downstreamJobs.each { folder ->
```

Comment:
A lot going on here, can probably be split into methods

Reply:
Agreed. Will split this up